### PR TITLE
add phpstan

### PR DESCRIPTION
**Pull request by mihaileu**

### CODE REVIEW

File: phpstan.neon.dist [hunk 1]
```diff
@@ -0,0 +1,9 @@
+parameters:
+    level: 5
+    checkMissingIterableValueType: false
+    paths:
+        - src
+        - tests
+    ignoreErrors:
+        - '#Access to an undefined property MongoDB\\Collection::\$documents.#'
+        - '#Return type \(void\) of method class@anonymous\/tests\/Unit\/MongoTransportTest.php:164::insertOne\(\) should be compatible with return type \(MongoDB\\InsertOneResult\) of method MongoDB\\Collection::insertOne\(\)#'
```
[the remark this answers is not in the document]
Comment (mihaileu) [reply phpstan.neon.dist:9]: sure, but let's take baby steps, and create issues for each case; at least this is my suggestion

File: composer.json [hunk 2]
```diff
@@ -41,7 +42,7 @@
     },
     "config": {
         "platform": {
-            "ext-mongo": "1.16.1"
+            "ext-mongodb": "1.7"
```
[the remark this answers is not in the document]
Comment (mihaileu) [reply composer.json:45]: This is how I obtain to run composer install without having mongo extension activated.
And platform is just a faker: https://getcomposer.org/doc/06-config.md#platform
I see no big deal of accepting this

[the remark this answers is not in the document]
Comment (iosifch) [reply composer.json:45]: Ah, I see. You're right!

File: phpstan.neon.dist (hunk 1 above)
[the remark this answers is not in the document]
Comment (iosifch) [reply phpstan.neon.dist:9]: I see. Ok, let's merge this pull request and I'll create an issue to solve the Phpstan errors.

**Pull request merged**